frontend/src/Components/ContentComponent.tsx

```diff
@@ -651,6 +651,40 @@ function ZoomComponent(props: ZoomComponentProps) {
     const defaultTranslateY = (window.innerHeight - props.height * defaultScale) / 2;
     useHotkeys('esc', props.onExit);
 
+    useEffect(() => {
+
+        // Event listener to handle back navigation
+        const handleBack = (event: PopStateEvent) => {
+            // console.log('Handling back event');
+            // preventing default back action if the new state isn't the popup
+            if (!event.state.popupOpen) {
+                // console.log('Preventing the default back action');
```

Comment:
You probably don't want to leave this in the final version.

Reply:
Removed and replaced with comments where applicable, thanks!